ext/random/zend_utils.c

```diff
@@ -0,0 +1,64 @@
+/*
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Arnaud Le Blanc <arnaud.lb@gmail.com>                       |
+   |          Tim Düsterhus <timwolla@php.net>                            |
+   +----------------------------------------------------------------------+
+*/
+
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#include "php_random_zend_utils.h"
+
+ZEND_ATTRIBUTE_NONNULL PHPAPI void php_random_bytes_insecure_for_zend(
+		zend_utility_random_bytes_insecure_state *opaque_state, void *bytes, size_t size)
+{
+	php_random_bytes_insecure_state_for_zend *state = (php_random_bytes_insecure_state_for_zend*) opaque_state;
+
+	if (UNEXPECTED(!state->initialized)) {
+		uint64_t t[4];
+		php_random_fallback_seed_state fallback_state;
+		fallback_state.initialized = false;
+
+		do {
+			/* Skip the CSPRNG if it has already failed */
+			if (!fallback_state.initialized) {
+				char errstr[128];
+				if (php_random_bytes_ex(&t, sizeof(t), errstr, sizeof(errstr)) == FAILURE) {
+#if ZEND_DEBUG
+					fprintf(stderr, "php_random_bytes_ex: Failed to generate a random seed: %s\n", errstr);
+#endif
+					goto fallback;
+				}
+			} else {
+	fallback:
```

Comment:
```suggestion
 fallback:
```
Small nit, not sure if there's any coding standard rules, but I prefer indenting labels with a single _space_.

This keeps them near the left of the file, making them stand out, while not looking like a function definition to naive diff tools.

Reply:
Ran a quick grep to find what we use most often. We are not super consistent, but we indent labels with tabs in 72% of the case in `Zend/` and `main/` when excluding generated files (170 labels indented with tabs, 19 with spaces, 46 without indentation).

Reply:
I've did a quick look with GitHub's search using `repo:php/php-src /^\s*[a-z]+:/ language:C`. Most of the results appear to be `default:`.

This raises the question: Did you exclude `default:` from your quick grep? Otherwise that might bias the results.

In any case, I don't care *too* strongly, but have a preference for the space 😄

Reply:
> This raises the question: Did you exclude `default:` from your quick grep? Otherwise that might bias the results.

Ah! There is indeed a lot of `default:`. After excluding them I get almost exclusively labels without indentation, except in zend_strtod.c, which indents labels with one space.